src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -334,52 +404,241 @@ private synchronized void writeToConsole(Person person, int index, long time, bo
     }
   }
 
-  private long setDemographics(Person person, Demographics city) {
-    person.attributes.put(Person.CITY, city.city);
-    person.attributes.put(Person.STATE, city.state);
-    
-    String race = city.pickRace(person.random);
-    person.attributes.put(Person.RACE, race);
-    String ethnicity = city.ethnicityFromRace((String)person.attributes.get(Person.RACE), person);
-    person.attributes.put(Person.ETHNICITY, ethnicity);
-    String language = city.languageFromEthnicity((String) person.attributes.get(Person.ETHNICITY),
-        person);
-    person.attributes.put(Person.FIRST_LANGUAGE, language);
-
-    String gender = city.pickGender(person.random);
-    if (gender.equalsIgnoreCase("male") || gender.equalsIgnoreCase("M")) {
-      gender = "M";
+  private long setDemographics(Person person, Demographics city) throws IOException {
+    if (householdsMode.equals("true")) {
+      // get a random spew person
+
+      int[] range = new int[] { 1, spewPerson.size() + 1 };
+
+      int randSpew = (int) person.rand(range);
+
+      String spewSerial = spewPerson.get(randSpew).get("SERIALNO");
```

Comment:
`spewPerson.get(randSpew)` is repeated a dozen times here - it should probably be extracted to a local variable